Add `Toast` documentation and implement close button customization features. (#283)
yiisoft/bootstrap5: docs/guide/en/toast.md

# Toast

The **Toast** widget renders a [toast](https://getbootstrap.com/docs/5.3/components/toasts/#examples) component.

You can use it to create lightweight notifications similar to push notifications used in mobile and desktop operating
systems. Toasts are built with flexbox and are easy to align and position.

## Key Features
- Customizable header with support for images, titles, and timestamps.
- Simple body content or custom complex content.
- Optional dismissable functionality with customizable close button.
- Trigger button integration for displaying toast notifications on demand.
- Flexible positioning through container utilities.

## Important Note

> Toasts are opt-in for performance reasons, so **you must initialize them with JavaScript** after rendering:

Reviewer: Can we do something about it?

1. I don't think it is bad for performance.
2. Registering the JS automatically would be great if possible.

> ## Important Note
> Toasts are opt-in for performance reasons, so **you must initialize them with JavaScript** after rendering:

Reviewer: ```suggestion
## Initializing and using toasts:

You must initialize and invoke toasts with JavaScript:
```

```javascript
// Initialize all toasts on the page
const toastElList = document.querySelectorAll('.toast');
const toastList = [...toastElList].map(toastEl => new bootstrap.Toast(toastEl));

// Or initialize a specific toast
const toastElement = document.getElementById('myToast');
const toast = new bootstrap.Toast(toastElement);

// Show the toast
toast.show();
```

## Quick Start
To get started, instantiate the **Toast** widget and customize it as needed. Here is a basic example:

```php
<?php

declare(strict_types=1);

use Yiisoft\Bootstrap5\Toast;
?>

<?= Toast::widget()
        ->body('Hello, world! This is a toast message.')
        ->image('path/to/image.jpg', 'Bootstrap logo', ['class' => 'rounded me-2'])
        ->time('11 mins ago', [], 'text-body-secondary')
        ->title('Bootstrap')
?>
```

This generates a simple toast notification with a header containing an image, title, timestamp, and a close button,
along with some body content.

## Configuration

### Setting Attributes
Customize the toast container with HTML attributes:

```php
Toast::widget()
    ->attributes(['id' => 'my-toast'])
    ->class('custom-toast');
```

> Add single attributes:

Reviewer: ```suggestion
Add single attribute:
```

Author: done.
